packages/react/src/lib/getOrGenerateIdFromShorthand.ts

```diff
@@ -6,20 +6,25 @@ const getOrGenerateIdFromShorthand = <P extends Record<string, any>>(
   prefix: string,
   value: ShorthandValue<P>,
   currentValue?: string,
+  fallbackToUnique?: boolean,
 ): string | undefined => {
-  if (_.isNil(value)) {
-    return undefined
-  }
+  let result: string
 
-  if (React.isValidElement(value)) {
-    return (value as React.ReactElement<{ id?: string }>).props.id
+  if (_.isNil(value)) {
+    result = undefined
+  } else if (React.isValidElement(value)) {
+    result = (value as React.ReactElement<{ id?: string }>).props.id
+  } else if (_.isPlainObject(value)) {
+    result = (value as Record<string, any>).id
+  } else {
+    result = currentValue || _.uniqueId(prefix)
   }
 
-  if (_.isPlainObject(value)) {
-    return (value as Record<string, any>).id
+  if (fallbackToUnique && !result) {
+    result = _.uniqueId(prefix)
   }
 
-  return currentValue || _.uniqueId(prefix)
+  return result
```

Comment:
Why this is needed? Please clarify.

As I see, this condition (`if (fallbackToUnique && !result) {)` can work only if `value` is `nil`...

Reply:
these changes are basically to make sure that an id is generated even if value is an element or shorthand without an ID.

Reply:
```tsx
import * as React from 'react'
import * as _ from 'lodash'
import { ShorthandValue } from '../types'

const getOrGenerateIdFromShorthand = <P extends Record<string, any>>(
  prefix: string,
  value: ShorthandValue<P>,
  currentValue?: string,
): string | undefined => {
  if (_.isNil(value)) {
    return undefined
  }

  let result: string

  if (React.isValidElement(value)) {
    result = (value as React.ReactElement<{ id?: string }>).props.id
  } else if (_.isPlainObject(value)) {
    result = (value as Record<string, any>).id
  }

  return result || currentValue || _.uniqueId(prefix)
}

export default getOrGenerateIdFromShorthand
```